cli/src/main/java/com/devonfw/tools/ide/tool/jasypt/Jasypt.java

```diff
@@ -0,0 +1,103 @@
+package com.devonfw.tools.ide.tool.jasypt;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.property.EnumProperty;
+import com.devonfw.tools.ide.property.PasswordProperty;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.tool.java.Java;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link ToolCommandlet} for <a href="http://www.jasypt.org/">Jasypt</a>, The java library which allows to add basic
+ * encryption capabilities with minimum effort.
+ */
+public class Jasypt extends LocalToolCommandlet {
+
+  public final EnumProperty<JasyptCommand> command;
+
+  public final PasswordProperty password;
+
+  public final PasswordProperty input;
+
+  private static final String CLASS_NAME_ENCRYPTION = "org.jasypt.intf.cli.JasyptPBEStringEncryptionCLI";
+
+  private static final String CLASS_NAME_DECRYPTION = "org.jasypt.intf.cli.JasyptPBEStringDecryptionCLI";
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Jasypt(IdeContext context) {
+
+    super(context, "jasypt", Set.of(Tag.JAVA, Tag.ENCRYPTION));
+
+    this.command = add(new EnumProperty<>("", true, "command", JasyptCommand.class));
+    this.password = add(new PasswordProperty("", true, "password"));
+    this.input = add(new PasswordProperty("", true, "input"));
```

Comment:
The parameters from jasypt are very generic and this makes it much more confusing for end-users:
We want to encrypt passwords with a master-password. If a parameter is just called password and both parameters are passwords which of those is it? I personally would prefer here to stay with devonfw-ide compatbility and the from my PoV better understandable parameters:
* `masterPassword` - the master password for encryption/decryption
* `secret` - the secret value to encrypt - e.g. a password

Reply:
Yess now that I better understand that normally is used to encrypt passwords, I totally agree 😄